atomic_reactor/plugins/post_tag_from_config.py

```diff
@@ -43,7 +44,51 @@ class TagFromConfigPlugin(PostBuildPlugin):
 
     TAGS_FILENAME = 'additional-tags'
 
-    def get_tags(self):
+    def __init__(self, tasker, workflow, tags=None):
+        super(TagFromConfigPlugin, self).__init__(tasker, workflow)
+        self.tags = tags
+
+    def parse_and_add_tags(self):
+        class MyFormatter(string.Formatter):
+            """
+            using this because str.format can't handle keys with dots and dashes
+            which are included in some of the labels, such as
+            'authoritative-source-url', 'com.redhat.component', etc
+            """
+            def get_field(self, field_name, args, kwargs):
+                return (self.get_value(field_name, args, kwargs), field_name)
+
+        tags = []
+        labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow,
+                           env_replace=True).labels
+
+        for tag in self.tags.get('unique', []):
+            try:
+                u_tag = MyFormatter().vformat(tag, [], labels)
+            except KeyError:
+                self.log.error('%s submitted as unique tag with no matching label', tag)
+                continue
+            if u_tag != tag:
+                self.log.error('%s submitted as unique tag', tag)
+                continue
+            self.log.debug('Using additional unique tag %s', u_tag)
+            self.workflow.tag_conf.add_unique_image(tag)
+            tags.append(tag)
+
+        if self.tags.get('primary', []):
+            for tag in self.tags.get('primary', []):
+                try:
+                    p_tag = MyFormatter().vformat(tag, [], labels)
+                except KeyError:
+                    self.log.error('%s submitted as primary tag with no matching label', tag)
+                    continue
```

Comment:
I wonder if it's better to just let this exception propagate rather than discarding the tag. If we have a bad tag here it means something is badly wrong with either the ATOMIC_REACTOR_PLUGINS config (bad tag template) or the Dockerfile (missing essential label).